techniques/android/MASTG-TECH-0109.md

```diff
@@ -0,0 +1,51 @@
+--- 
+title: Intercepting Flutter HTTP Traffic
+platform: android 
+---
+
+Flutter is an open-source UI software development kit (SDK) created by Google. It is used for building natively compiled applications for mobile, web, and desktop from a single codebase. Flutter uses Dart, which is not proxy-aware and uses its own certificate store. The application doesn't take proxy configuration from the system and send the data directly to the server. Due to this, it is not possible to intercept the request using the BurpSuite or any MITM tools.
+
+
+
+**How does re-flutter method differs from other techniques ?**
+
+There are alternative methods for intercepting traffic, such as [sending traffic to the proxy through ProxyDroid/iptables](https://blog.nviso.eu/2019/08/13/intercepting-traffic-from-android-flutter-applications/). However, these techniques require some configuration. By employing the re-flutter command-line tool, the application can be patched effortlessly without the need for any setup.
```

Comment:
```suggestion
In order to intercept Flutter HTTPS traffic we need to deal with two problems:
* Make sure the traffic is sent to the proxy
* Disable the TLS verification of any HTTPS connection

There are generally two approaches to this: reFlutter and Frida. 

* reFlutter: This tool creates a modified version of libFlutter.so which is then repackaged into the APK. It configures the internal libraries to use a specified proxy and disable the TLS verification
* Frida: The [disable-flutter-tls.js script](https://github.com/NVISOsecurity/disable-flutter-tls-verification) can dynamically remove the TLS verification without the need for repackaging. As it doesn't modify the proxy configuration, additional steps are needed (e.g. ProxyDroid, DNS, iptables, ...)
```
The re-flutter app also has downsides, so this is pretty one-sided:
* Only works on known/published hashes, otherwise you have to rebuild the engine yourself
* Repackages the app. This brings complications (e.g. iOS or any app that detects repackaging)

I've used reFlutter a few times (mostly for the object-dump) and it's great when it works, but not straightforward if it doesn't.

So I modified this section to give a more generic introduction of what needs to be done, and then the rest can explain both reFlutter and Frida+(eg)ProxyDroid

We should also use reFlutter, as the tool calls itself, and not re-flutter.